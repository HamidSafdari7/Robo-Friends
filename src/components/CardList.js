import React from 'react'
import Card from './Card';

const CardList = (props) => {
    const { robots } = props;

    return (
        <>
            <div>

                {
                    robots.map((user, index) => {

                        return (
                            <Card
                                key={index}
                                id={robots[index].id}
                                name={user.name}
                                username={user.username}
                                email={user.email}
                            />
                        );
                    })
                }

            </div>
        </>
    )
}

export default CardList
