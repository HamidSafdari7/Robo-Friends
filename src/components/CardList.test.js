import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'Cristiano Ronaldo',
            username: 'CR7',
            email: 'cr7@gmail.com'
        }
    ]
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
})