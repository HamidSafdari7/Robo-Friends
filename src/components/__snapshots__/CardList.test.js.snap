// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<Fragment>
  <div>
    <Card
      email="cr7@gmail.com"
      id={1}
      key="0"
      name="Cristiano Ronaldo"
      username="CR7"
    />
  </div>
</Fragment>
`;
